rust/cardano-blockchain-types/src/cip36/mod.rs

```diff
@@ -0,0 +1,143 @@
+//! CIP-36 Catalyst registration module
+
+pub mod key_registration;
+pub mod registration_witness;
+mod validation;
+pub mod voting_pk;
+
+use ed25519_dalek::VerifyingKey;
+use key_registration::Cip36KeyRegistration;
+use pallas::ledger::addresses::ShelleyAddress;
+use registration_witness::Cip36RegistrationWitness;
+use validation::{validate_payment_address_network, validate_signature, validate_voting_keys};
+use voting_pk::VotingPubKey;
+
+use crate::{MetadatumValue, Network};
+
+/// CIP-36 Catalyst registration
+#[derive(Clone, Default, Debug)]
+pub struct Cip36 {
+    /// Key registration - 61284
+    key_registration: Cip36KeyRegistration,
+    /// Registration witness - 61285
+    registration_witness: Cip36RegistrationWitness,
+    /// Is this a Catalyst strict registration?
+    is_catalyst_strict: bool,
+}
+
+/// Validation value for CIP-36.
+#[allow(clippy::struct_excessive_bools, clippy::module_name_repetitions)]
+#[derive(Clone, Default, Debug)]
+pub struct Cip36Validation {
+    /// Is the signature valid? (signature in 61285)
```

Comment:
This is not a sufficient error report, there are lots of other errors, which is why we had a human readable error report before.